tests/foreman/api/test_setting.py

```diff
@@ -108,9 +107,21 @@ def test_negative_update_hostname_with_empty_fact(self):
 
         :expectedresults: Error should be raised on setting empty value for
             hostname_facts setting
-
-        :CaseAutomation: notautomated
         """
+        hostname_facts_id = [
+            ele.id
+            for ele in entities.Setting().search(
+                query={"per_page": 200, 'search': 'name="discovery_hostname"'}
+            )
+        ][0]
```

Comment:
If you only going to use first value, you might as well not iterate over all of them:
```suggestion
        hostname_facts_id = entities.Setting().search(
            query={"per_page": 200, 'search': 'name="discovery_hostname"'}
        )[0].id
```